student/forms.py

```diff
@@ -0,0 +1,37 @@
+from django import forms
+from django.contrib.auth.forms import UserCreationForm
+from django.contrib.auth.models import User
+from .models import Student, EducationalInstitution, Major
+
+
+class studentRegistrationForm(UserCreationForm):
+    educationalInstitution_Queryset = EducationalInstitution.objects.all()
+    full_name = forms.CharField(max_length=255)
+    email = forms.EmailField(required=True)
+    date_of_birth = forms.DateField()
+    phone_number = forms.CharField(max_length=15)
+    educational_institution = forms.ModelChoiceField(
+        educationalInstitution_Queryset)
+    major = forms.TypedChoiceField(choices=Major.choices)
+    about = forms.CharField(widget=forms.Textarea)
+    graduation_date = forms.DateField()
+
+    class Meta1:
+        model = User
+        fields = ['username', 'email', 'password1', 'password2']
+
+    def save(self, commit=True):
+        user = super(studentRegistrationForm, self).save(commit=False)
+        user.email = self.cleaned_data['email']
+        if commit:
+            user.save()
+        user2 = Student(user=user,
+                        full_name=self.cleaned_data['full_name'],
+                        date_of_birth=self.cleaned_data['date_of_birth'],
+                        phone_number=self.cleaned_data['phone_number'],
+                        educational_institution=self.cleaned_data['educational_institution'],
+                        major=self.cleaned_data['major'],
+                        about=self.cleaned_data['about'],
+                        graduation_date=self.cleaned_data['graduation_date'])
+        user2.save()
```

Comment:
why are you overriding the default `save()` method?

Reply:
what do you mean? that is the way I saw online to save your own form In several different sites and tutorials. 
there is a better or different way to do it?

Reply:
The `save` method is a built-in method in to save an instance into a particular model (in this case student), you don't need to override it unless you have some special constrains that you want to add. Here it seems that your function saves the user 3 times in the DB. why? 

@ifireball thoughts?

Reply:
if I understand correctly this technique user saves the data into the user's table, user2 saves that data into the student's table, the code in line 24 is if commit=false, but the second user. save is if the user sent the form correctly and just then save him. you can look at the discussion here?  
https://stackoverflow.com/a/37820531
how would you do in a different @guy9050? maybe a short example will help. :)

Reply:
I saw some tutorials online on how to do this thing of a single form for 2 tables. and that worked good, I don't think that needs to be changed. that decision for making this partition in the registration form came after long thinking in our team meeting.  we want to keep it this way.